apisix/init.lua

```diff
@@ -534,6 +533,10 @@ end
 
 
 function _M.http_header_filter_phase()
+    if not ngx.ctx or not ngx.ctx.api_ctx then
+        ngx.ctx = ctxdump.apply_ngx_ctx(ngx_var.ctx_ref)
```

Comment:
Stash `ngx.ctx` unconditionally but checking the existence of `ngx.ctx` , this behavior leaks some keys for `ngx.ctx` tables (for those requests which doesn't make internal redirect).

Reply:
Fixed. Please review it again.